fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -4160,7 +4164,42 @@ public void alterTableProperties(Database db, OlapTable table, Map<String, Strin
                                 ImmutableMap.of(key, propertiesToPersist.get(key)));
                 GlobalStateMgr.getCurrentState().getEditLog().logAlterTableProperties(info);
             }
+            if (propertiesToPersist.containsKey(PropertyAnalyzer.PROPERTIES_DATACACHE_ENABLE)) {
+                String dataCacheEnable = propertiesToPersist.get(PropertyAnalyzer.PROPERTIES_DATACACHE_ENABLE);
+                boolean isEnable = Boolean.parseBoolean(dataCacheEnable);
 
+                if (table instanceof LakeTable) {
+                    LakeTable lakeTable = (LakeTable) table;
+                    GlobalStateMgr.getCurrentState().getStarOSAgent().updateShardGroup(lakeTable, isEnable);
```

Comment:
this will throw StarClientException, what will end user see if excepted here?

Reply:
updateShardGroup will throw DdlException if error occurs.
The following picture shows what happen in shared-nothing mode if we want to alter datacache.enable
![image](https://github.com/StarRocks/starrocks/assets/145319350/f1781b2c-f437-47b4-ae2c-3a8f75570652)